tests/manage/test_pvresize_with_node_down.py

```diff
@@ -0,0 +1,228 @@
+    setup_cmd = f'rsh {OCS_BUG_ID}-pod {real_cmd}'
+    full_cmd = f'oc -n {config.ENV_DATA["my_namespace"]} {setup_cmd}'
+    not_run = True
+    fcount = 0
+    while not_run:
+        try:
+            run_cmd(full_cmd)
+            not_run = False
+        except CommandFailed as ex:
+            log.info(f'{ex} raised on {setup_cmd}')
+            fcount += 1
+            if fcount > FAIL_LIMIT:
+                raise CommandFailed(f'Too many failures: {setup_cmd}')
+            sleep(5)
+
+
+def test_add_stuff_to_rbd():
+    """
+    Make sure that there is data in the rbd pool.
+
+    The /mnt/{OCS_BUG_ID} diretory on the app pod is mounted on a
+    block device that is implemented by openshift-storage rbd.
+    Writing here will add data to that storage.
+    """
+    df_info = CEPH_TOOL.exec_ceph_cmd('ceph df')
+    prev_avail = df_info['stats']['total_avail_bytes']
+    BIG_FILE = '/usr/lib/x86_64-linux-gnu/libicudata.so.57.1'
+    for i in range(0, 10):
+        param = f'cp {BIG_FILE} /mnt/{OCS_BUG_ID}/xx.{i}'
+        cmd_on_my_namespace(param)
+    df_info = CEPH_TOOL.exec_ceph_cmd('ceph df')
+    now_avail = df_info['stats']['total_avail_bytes']
+    assert now_avail < prev_avail
+
+
+def update_pvc():
+    """
+    Update the size of the pvc.
+
+    First modify the storage class to allow expansion.
+    Then read the pv so that we are sure that the change has taken effect
+    before increasing the size of the pvc
+    """
+    NEW_SIZE = '200Gi'
+    cmd_front = f"oc -n {config.ENV_DATA['my_namespace']} patch"
+    cmd1_middle = 'storageclass/"rook-ceph-block"'
+    cmd2_middle = 'pvc/ocs-pvc'
+    cmd_1 = ' '.join([cmd_front, cmd1_middle, '--patch'])
+    cmd_1 += " '{\"allowVolumeExpansion\": true}'"
+    run_cmd(cmd_1)
+    run_cmd('oc get pv')
+    cmd_2 = ' '.join([cmd_front, cmd2_middle, '--patch'])
+    cmd_2 += " '{\"spec\": {\"resources\": {\"requests\": {\"storage\": \""
+    cmd_2 += NEW_SIZE
+    cmd_2 += "\"}}}}'"
+    run_cmd(cmd_2)
+
+
+@tier1
+@pytest.mark.usefixtures(
+    test_fixture.__name__,
+)
+@pytest.mark.polarion_id("OCS-368")
+class TestOcs368(ManageTest):
+    """
+    Actual OCS-368 test.
+    """
+    def test_ocs_368(self):
+        """
+        The test_add_stuff_to_rbd call makes sure that there is data written
+        to the rbd pool
+
+        Two threads are started.  One thread removes an OSD.  A new OSD
+        will then be started by Rook.  The other thread modifies the
+        pvc size for the pvc corresponding to the rbd pool
+
+        After both thread complete, we wait until all osds are up.
+
+        After all OSDs are up, we wait until ceph is HEALTHY
+        """
+        test_add_stuff_to_rbd()
+        pvc_chg = threading.Thread(target=update_pvc)
+        pvc_chg.start()
+        cmd_str = 'delete deployment.apps/rook-ceph-osd-2'
+        cmd_str = f'oc -n {config.ENV_DATA["cluster_namespace"]} {cmd_str}'
+        run_cmd(cmd_str)
+        pvc_chg.join()
+        while True:
```

Comment:
I am adding a check to make sure this  will terminate after an hour.

Reply:
An hour is far too long for one test to be spinning it's wheels in a loop like this. Let's determine a more reasonable maximum time to loop for.

Reply:
I think that this could be reduced into one loop.  All we care about is being healthy at the end.  So I'm probably going to rework this a bit.

The last test that I ran took a little over 20 minutes:

23:48:39 - Dummy-12 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ec2-user/wusui/auth/kubeconfig get PersistentVolumeClaim -A -o yaml
23:48:39 - Dummy-13 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ec2-user/wusui/auth/kubeconfig get Secret -A -o yaml
23:48:39 - Dummy-14 - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/ec2-user/wusui/auth/kubeconfig get Namespace -A -o yaml


================== 1 passed, 1 deselected in 1307.63 seconds ===================

So maybe just one loop for 20 minutes would be good? (some of the testing consumed time before I even got into this loop).

Reply:
The next version will have one 20 minute maximum loop

Reply:
Consider moving the logic in this loop to a function that raises an exception when the health check fails. This way you can decorate it with the `retry` decorator in `ocs_ci.utility.retry` instead of implementing your own retry logic. Here's an [example](https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/test_remove_mon_from_cluster.py#L24) of it being used. Also the pydoc for the decorator should be useful.

Reply:
I will use a retry decorator.